Add Quickstart documentation for Docker deployments (#209)
elastic/opentelemetry: docs/quickstart/serverless/docker.md

2. Retrieve the `Elastic OTLP Endpoint` and the `Elastic API Key` for your Serverless Project by [following these instructions](./#retrieving-connection-details-for-your-serverless-project).

3. Create a `.env` file with the following content, replacing the placeholder values with your actual Elastic Cloud credentials:

   ```bash
   HOST_FILESYSTEM=/
   DOCKER_SOCK=/var/run/docker.sock
   ELASTIC_AGENT_OTEL=true
   COLLECTOR_CONTRIB_IMAGE=elastic/elastic-agent:9.0.0
   ELASTIC_API_KEY=<your_api_key_here>
   ELASTIC_OTLP_ENDPOINT=<your_endpoint_here>
   OTEL_COLLECTOR_CONFIG=/path/to/otel-collector-config.yml
   ```

4. Create a `compose.yml` file with the following content:

   ```yaml
   services:
    otel-collector:
      image: ${COLLECTOR_CONTRIB_IMAGE}
      container_name: otel-collector
      deploy:
        resources:
          limits:
            memory: 1.5G
      restart: unless-stopped
      command: ["--config", "/etc/otelcol-config.yml" ]
      network_mode: host
      user: 0:0
      volumes:
        - ${HOST_FILESYSTEM}:/hostfs:ro
        - ${DOCKER_SOCK}:/var/run/docker.sock:ro
        - ${OTEL_COLLECTOR_CONFIG}:/etc/otelcol-config.yml

>       environment:

Reviewer: Same here (ELASTIC_AGENT_OTEL)

   ```bash
   HOST_FILESYSTEM=/
   DOCKER_SOCK=/var/run/docker.sock
   ELASTIC_AGENT_OTEL=true
   COLLECTOR_CONTRIB_IMAGE=elastic/elastic-agent:9.0.0
   ELASTIC_API_KEY=<your_api_key_here>
   ELASTIC_OTLP_ENDPOINT=<your_endpoint_here>
   OTEL_COLLECTOR_CONFIG=/path/to/otel-collector-config.yml
   ```

4. Create a `compose.yml` file with the following content:

   ```yaml
   services:
    otel-collector:
      image: ${COLLECTOR_CONTRIB_IMAGE}
      container_name: otel-collector
      deploy:
        resources:
          limits:
            memory: 1.5G
      restart: unless-stopped
      command: ["--config", "/etc/otelcol-config.yml" ]
      network_mode: host
      user: 0:0
      volumes:
        - ${HOST_FILESYSTEM}:/hostfs:ro
        - ${DOCKER_SOCK}:/var/run/docker.sock:ro
        - ${OTEL_COLLECTOR_CONFIG}:/etc/otelcol-config.yml
      environment:
        - HOST_FILESYSTEM
        - ELASTIC_AGENT_OTEL
        - ELASTIC_API_KEY
        - ELASTIC_OTLP_ENDPOINT
        - STORAGE_DIR=/usr/share/elastic-agent
   ```

5. Start the collector by running:

   ```bash
   docker compose up -d
   ```

6. **(Optional) Instrument your applications**

>     If you would like to collect telemetry from applications running on the host where you installed the EDOT Collector,
>     you need to instrument your target applications according to the setup instructions for corresponding EDOT SDKs:

Reviewer 2: ```suggestion
 To collect telemetry from applications running on the host where you installed the EDOT Collector, 
  instrument your target applications following the setup instructions for the corresponding language:
```
